example/client/client.go

```diff
@@ -9,29 +9,37 @@ import (
 	"io/ioutil"
 	"log"
 	"net/http"
-	"net/http/httptrace"
-	"time"
+	"os"
 
-	"github.com/honeycombio/opentelemetry-exporter-go/honeycomb"
-
-	otelhttptrace "go.opentelemetry.io/contrib/instrumentation/net/http/httptrace/otelhttptrace"
-	otelhttp "go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp"
-	"go.opentelemetry.io/otel/api/baggage"
+	"go.opentelemetry.io/contrib/instrumentation/net/http/httptrace/otelhttptrace"
+	"go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp"
+	"go.opentelemetry.io/otel"
 	"go.opentelemetry.io/otel/api/global"
 	"go.opentelemetry.io/otel/api/trace"
 	"go.opentelemetry.io/otel/label"
+	"go.opentelemetry.io/otel/propagators"
 	sdktrace "go.opentelemetry.io/otel/sdk/trace"
 	"go.opentelemetry.io/otel/semconv"
+
+	"github.com/honeycombio/opentelemetry-exporter-go/honeycomb"
 )
 
-func initTracer(exporter *honeycomb.Exporter) {
-	// For the demonstration, use sdktrace.AlwaysSample sampler to sample all traces.
-	// In a production application, use sdktrace.ProbabilitySampler with a desired probability.
-	tp := sdktrace.NewTracerProvider(
-		sdktrace.WithConfig(sdktrace.Config{DefaultSampler: sdktrace.AlwaysSample()}),
-		sdktrace.WithSyncer(exporter),
-	)
+func initTracer(exporter *honeycomb.Exporter) func() {
+	bsp := sdktrace.NewBatchSpanProcessor(exporter)
+	tp := sdktrace.NewTracerProvider(sdktrace.WithSpanProcessor(bsp))
+	tp.ApplyConfig(
+		// The default takes parent sampler hints into account, which we don't need here.
+		sdktrace.Config{
+			// For the demonstration, use sdktrace.AlwaysSample sampler to sample all traces.
+			// In a production application, use sdktrace.ProbabilitySampler with a desired
+			// probability.
+			DefaultSampler: sdktrace.AlwaysSample(),
+		})
 	global.SetTracerProvider(tp)
+	global.SetTextMapPropagator(otel.NewCompositeTextMapPropagator(
+		propagators.TraceContext{},
+		propagators.Baggage{}))
+	return bsp.Shutdown
```

Comment:
See open-telemetry/opentelemetry-specification#1073 for how this situation might change soon. My comment https://github.com/open-telemetry/opentelemetry-specification/issues/1073#issuecomment-706458748 addresses my use of `NewBatchSpanProcessor` above, as opposed to using `WithBatcher`.

Reply:
open-telemetry/opentelemetry-go#1227 just merged, allowing shutting down a `TracerProvider`, in trun shutting down its registered span processors. Once we have release available that includes that new method, I'll propose an update to simplify this example.